src/structures/lists/pcr_selection.rs

```diff
@@ -0,0 +1,223 @@
+
+        if size > PcrSelectionList::MAX_SIZE {
+            error!(
+                "Invalid size value in TPML_PCR_SELECTION (> {})",
+                PcrSelectionList::MAX_SIZE
+            );
+            return Err(Error::local_error(WrapperErrorKind::InvalidParam));
+        }
+
+        let mut items = HashMap::<HashingAlgorithm, PcrSelection>::new();
+        // Loop over available selections
+        for tpms_pcr_selection in tpml_pcr_selection.pcrSelections[..size].iter() {
+            // Parse pcr selection.
+            let parsed_pcr_selection = PcrSelection::try_from(*tpms_pcr_selection)?;
+            // Insert the selection into the storage. Or merge with an existing.
+            match items.get_mut(&parsed_pcr_selection.hashing_algorithm()) {
+                Some(previously_parsed_pcr_selection) => {
+                    previously_parsed_pcr_selection.merge(&parsed_pcr_selection)?;
+                }
+                None => {
+                    let _ = items.insert(
+                        parsed_pcr_selection.hashing_algorithm(),
+                        parsed_pcr_selection,
+                    );
+                }
+            }
+        }
+        Ok(PcrSelectionList { items })
+    }
+}
+
+/// A builder for the PcrSelectionList struct.
+#[derive(Debug, Default)]
+pub struct PcrSelectionListBuilder {
+    size_of_select: Option<PcrSelectSize>,
+    items: HashMap<HashingAlgorithm, BitFlags<PcrSlot>>,
+}
+
+impl PcrSelectionListBuilder {
+    pub fn new() -> Self {
+        PcrSelectionListBuilder {
+            size_of_select: None,
+            items: Default::default(),
+        }
+    }
+
+    /// Set the size of the pcr selection(sizeofSelect)
+    ///
+    /// # Arguments
+    /// size_of_select -- The size that will be used for all selections(sizeofSelect).
+    pub fn with_size_of_select(mut self, size_of_select: PcrSelectSize) -> Self {
+        self.size_of_select = Some(size_of_select);
+        self
+    }
+
+    /// Adds a selection associated with a specific HashingAlgorithm.
+    ///
+    /// This function will not overwrite the values already associated
+    /// with a specific HashingAlgorithm only update.
+    ///
+    /// # Arguments
+    /// hash_algorithm -- The HashingAlgorithm associated with the pcr selection
+    /// pcr_slots      -- The PCR slots in the selection.
+    /// size_of_select --
+    pub fn with_selection(
+        mut self,
+        hash_algorithm: HashingAlgorithm,
+        pcr_slots: &[PcrSlot],
+    ) -> Self {
+        let selected_pcr_slots: BitFlags<PcrSlot> = pcr_slots.iter().cloned().collect();
+        match self.items.get_mut(&hash_algorithm) {
+            Some(previously_selected_pcr_slots) => {
+                *previously_selected_pcr_slots |= selected_pcr_slots;
+            }
+            None => {
+                let _ = self.items.insert(hash_algorithm, selected_pcr_slots);
+            }
+        }
+        self
+    }
```

Comment:
Should there be a limit here, to check if `self.items` contains more than `PcrSelectionList::MAX_SIZE`?

Reply:
Hrrmm...
Yes, I think there should be.

Reply:
Or rather it should be but I am not sure it will have any effect. Because there are currently only eight items in the HashingAlgorithm enum. And the MAX_SIZE is 16. And because they are squashed together it is never possible to create a tss list with a count greater then 8.

Reply:
> Because there are currently only eight items in the HashingAlgorithm enum. And the MAX_SIZE is 16.

Ha, yes, I didn't realise that, good point!